week-2-wide-perspective/w02e02-data-manipulation/src/data.js

```diff
@@ -0,0 +1,93 @@
+import axios from 'axios'
+import { Subject } from 'rxjs'
+import { v4 } from 'uuid'
+
+export const STATUS = {
+    loading: 'LOADING',
+    success: 'SUCCESS',
+    fail: 'FAI',
+}
+
+const dummyDelay = request => new Promise((resolve, reject) => {
```

Comment:
Można to strumieniem obsłużyć, zobacz `timer` oraz `switchMap`
https://www.learnrxjs.io/learn-rxjs/operators/creation/timer

https://www.learnrxjs.io/learn-rxjs/operators/transformation/switchmap

```javascript
timer(1000).pipe(switchMap(() => dataSubject))
```